hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/LazyAttributesMetadata.java

```diff
@@ -49,7 +48,7 @@ public static LazyAttributesMetadata from(
 					property,
 					isEnhanced,
 					allowEnhancementAsProxy,
-					collectionsInDefaultFetchGroupEnabled
+					false
```

Comment:
Just curious: what was the reason for this change?

Reply:
We have to consider attributes that are lazy from a persistence perspective, as lazy in bytecode enhancement. The flag would have caused the bytecode enhancement to see collections as part of the base fetch group which more or less means initialized. This is why the size code was triggered.

The whole bytecode enhancement logic is a bit of a mess in my opinion and I plan to improve this significantly in 6.0.

Reply:
@beikov well if that's the case then that sounds like we should be rolling back my patch which added `collectionsInDefaultFetchGroupEnabled` in the first place! Apparently it only worked by coincidence because there was another bug.

The semantics of `collectionsInDefaultFetchGroupEnabled` was _supposed_ to be:

1. that the field would be initialized, but
2. that the collection wrapper would remain uninitialized and unfetched.

It was definitely _not_ supposed to mean that the collection would be eagerly fetched.

But please define what you mean by "more or less" here.

Reply:
By not considering this flag here, it gets your desired semantics.

This part of the code determines if an attribute is considered `lazy`, which bytecode enhancement uses for determining whether it needs to initialize the object on access. Any attributes that are not listed as lazy are considered to be initialized. So when we access the collection then through calling `size()` this causes lazy initialization.

Reply:
FTR: @beikov and I chatted on the phone, and the decided that the changes related to `collectionsInDefaultFetchGroupEnabled` were not necessary.